editor/editor_spin_slider.cpp

```diff
@@ -102,6 +102,7 @@ void EditorSpinSlider::_gui_input(const Ref<InputEvent> &p_event) {
 			grabbing_spinner_dist_cache += diff_x;
 
 			if (!grabbing_spinner && ABS(grabbing_spinner_dist_cache) > 4 * EDSCALE) {
+				grabbing_spinner_mouse_pos = get_global_mouse_position();
```

Comment:
I would move this to after line 72, in the logic which handles the spinner being clicked.

When the mouse_pos is set here on line 105, when I let go of my mouse button the mouse does not appear _exactly_ where I pressed down, it can be quite a bit off depending on how I move my mouse.

This is because of the `ABS(grabbing_spinner_dist_cache) > 4 * EDSCALE` code. `dist_cache` is made up of motion on the x axis only. So if you move your mouse on the y axis first, and then on the x, that position of the mouse is what will be saved, which is not actually the location of the mouse when you first clicked down.

Observe: I move my mouse down only first, then on the x, then let go and my mouse does not reappear where it should be.

https://user-images.githubusercontent.com/41730826/119246767-9190bd00-bbc7-11eb-966b-ff0b45191458.mp4

Reply:
The question is: 
Where do you want your mouse pointer to reappear? 
Option A: from where it disappeared
Option B: from where you started dragging

I made the PR for Option A. But indeed I see the reasons for preferring Option B.

Reply:
My issue was is that it was inconsistent. If I dragged 15 units down and 4 units across very fast to change the value, I might not have realised I dragged 15 units down, and yet my mouse cursor would re-appear there.

I guess it could be personal preference - I just found that when I tested the original code it was a bit weird how the cursor did not return to where I initially clicked to initiate the value change.